### PR TITLE
Fix issue 377

### PR DESCRIPTION
**Pull request by idhamari**
New implementation to solve [Issue 377:  Enable definition of traning and validation set for privacy-preserving machine learning ](https://github.com/arx-deidentifier/arx/issues/377).

There is still a bug, probably related to [this line]( https://github.com/iaBIH/arx/blob/a90d45735a36354d64cbd8135a7f33ba5edacfe4/src/main/org/deidentifier/arx/aggregates/StatisticsClassification.java#L842):

            DataSubset subsetTrain = inputHandle.getSubset();
It seems the returned value is null when the handle comes from the GUI.

### CODE REVIEW

**Comment by prasser** [comment]
Please take a look at https://github.com/arx-deidentifier/arx/tree/feature-training-test

As mentioned, I merged your other PR into this branch and revised it. feature-training-test is the currently leading implementation of the feature, so if you want to further investigate this issue, please base your efforts on this branch. Will close this.

**Comment by idhamari** [comment]
Thanks for your feedback. I just tested the mentioned branch and I don't see the option for selecting kfold or subset for evaluation.  I will add the UI part based on your recent updates.

**Comment by prasser** [comment]
The setting is there. It is contained in the project properties dialog.

**Comment by idhamari** [comment]
ok, I found the option "Use test and training set:" in Edit/Settings/Utility analysis.
It seems it stopped at

**Comment by prasser** [comment]
Its not an error but expected behavior. You need to click the button to show the overall dataset.

**Comment by idhamari** [comment]
Thanks, everything works now after clicking the "Toggle sample view" button, maybe it should be activated if "Use test and training set" is selected!
Screenshot from 2022-06-20 16-42-36
